extensions/wikia/AdEngine/js/video/ooyalaAdSetProvider.js

```diff
@@ -0,0 +1,69 @@
+/*global define*/
+define('ext.wikia.adEngine.video.ooyalaAdSetProvider', [
+	require.optional('ext.wikia.adEngine.adContext'),
+	require.optional('ext.wikia.adEngine.video.vastUrlBuilder')
+], function (adContext, vastUrlBuilder) {
+	'use strict';
+
+	var FEATURED_VIDEO_SIZE = 640 / 480;
+
+	function generateSet(vpos, type, position, rv, correlator) {
+		return {
+			position_type: type,
+			position: position,
+			tag_url: vastUrlBuilder.build(FEATURED_VIDEO_SIZE, {
+				pos: 'FEATURED',
+				src: 'premium',
+				rv: rv
+			}, {
+				vpos: vpos,
+				correlator: correlator
+			})
+		};
+	}
+
+	function calculateRV(videoDepth, capping) {
+		return videoDepth === 1 ? 1 : Math.floor((videoDepth - 1) / capping) + 1;
```

Comment:
It works like rv for FMR refreshing, it resets every page view and it's increased every ad show 

- open page
- show preroll ad (rv=1)
- video
- (no preroll for this video)
- next video
- show preroll ad (rv=2)
- refresh page
- show preroll ad (rv=1)